lib/backends/sdc/containers.js

```diff
@@ -981,130 +999,387 @@ function renameLinks(opts, newName, callback) {
+            memory: common.humanSizeFromBytes(memory * (1024 * 1024))
+        }));
+        return;
+    }
 
-                if (!pkg.active) {
-                    log.info({
-                        account_uuid: opts.account.uuid,
-                        candidate: pkg
-                    }, 'skipping candidate because package is not active');
-                    return;
-                }
+    log.debug({pkg: candidate}, 'selected package for VM');
+    callback(null, candidate.uuid);
+}
 
-                if (pkg.max_physical_memory >= memory) {
-                    if (!candidate.hasOwnProperty('max_physical_memory')) {
-                        candidate = pkg;
-                        log.trace({
-                            account: opts.account.uuid,
-                            candidate: candidate,
-                            target_memory: memory
-                        }, 'initial candidate');
-                    } else if (pkg.max_physical_memory
-                        < candidate.max_physical_memory) {
-
-                        candidate = pkg;
-                        log.trace({
-                            account: opts.account.uuid,
-                            candidate: candidate,
-                            target_memory: memory
-                        }, 'new candidate');
-                    }
-                } else {
-                    log.trace({
-                        account: opts.account.uuid,
-                        candidate: pkg,
-                        target_memory: memory
-                    }, 'unacceptable candidate');
-                }
-            });
+/*
+ * Loops through the pkgs and tries to find the one that maches specified
+ * package one of the following ways:
+ *
+ *  - if specifiedPackage is a UUID, then pkg.uuid must match exactly
+ *  - if specifiedPackage is a package name, the pkg.name must match exactly
+ *  - if neither ^^ are true and the specifiedPackage is a Short-UUID, use the
+ *    package it shortened.
+ *
+ * It only considers pkgs that:
+ *
+ *  - are owned by nobody or this user
+ *  - are active
+ *
+ * The callback is called:
+ *
+ *   callback(err, package_uuid)
+ *
+ * Where err is an Error object if no package was found that matches the
+ * criteria, and package_uuid is passed if a package was found.
+ */
+function getSpecifiedPackage(opts, pkgs, specifiedPackage, callback) {
+    assert.object(opts, 'opts');
+    assert.object(opts.log, 'opts.log');
+    assert.array(pkgs, 'pkgs');
+    assert.string(specifiedPackage, 'specifiedPackage');
+    assert.func(callback, 'callback');
 
-            if (!candidate.uuid) {
-                callback(new errors.NoSufficientPackageError(constraints));
-                return;
+    var constraint = {};
+    var foundUuid;
+    var log = opts.log;
+    var nameMatches = [];
+    var uuidMatches = [];
+
+    constraint[PACKAGE_SELECTION_LABEL] = specifiedPackage;
+
+    if (pkgs.length === 1) {
```

Comment:
It seems it would be more consistent to check that in the case where `pkgs.length === 1`, `pkgs[0]` actually matches `specificiedPackage`. Thus, it might be more consistent to not treat that as a special case.